homeassistant/components/mqtt/__init__.py

```diff
@@ -760,6 +760,8 @@ def _mqtt_on_connect(
 
     def _mqtt_on_message(self, _mqttc, _userdata, msg) -> None:
         """Message received callback."""
+        # Add job because the subscription callback may call back into the
+        # MQTT client, which will deadlock
```

Comment:
how can the callback call back into MQTT client? Publish is done with a create task ?